core/controllers/acl_decorators_test.py

```diff
@@ -1216,6 +1216,52 @@ def test_normal_user_can_manage_preferences(self):
         self.logout()
 
 
+class ManageOwnProfileTests(test_utils.GenericTestBase):
+    """Tests for decorator can_manage_own_profile."""
+
+    banned_user = 'banneduser'
+    banned_user_email = 'banned@example.com'
+    username = 'user'
+    user_email = 'user@example.com'
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+        @acl_decorators.can_manage_own_profile
+        def get(self):
+            return self.render_json({'success': 1})
+
+    def setUp(self):
+        super(ManageOwnProfileTests, self).setUp()
+        self.signup(self.banned_user_email, self.banned_user)
+        self.signup(self.user_email, self.username)
+        self.set_banned_users([self.banned_user])
+        self.mock_testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock/', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+
+    def test_banned_user_cannot_update_profile_preferences(self):
+        self.login(self.banned_user_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json('/mock/', expected_status_int=401)
```

Comment:
Suggest adding a brief comment explaining why the status check 401 is the assertion to verify that the behavior in this case is correct.

Reply:
Does this seem okay?
```python
    def test_banned_user_cannot_update_profile_preferences(self):
        self.login(self.banned_user_email)
        
        # Expected status check 401 because a banned user does not have
        # the rights to manage a profile.
        with self.swap(self, 'testapp', self.mock_testapp):
            self.get_json('/mock/', expected_status_int=401)
        self.logout()
```

Reply:
Yup, LGTM.

Reply:
Done.